main.cpp

```diff
@@ -121,369 +155,446 @@ struct CarWash
+    {
+        tapeReadSpeed *= tapeSpeed;
+    }
+    else
+    {
+        tapeReadSpeed = 1;
+    }
+}
 
 struct SolideStateAmp
 {
-    //number of speakers
     int numSpeakers = 2;
-    //speaker size in inch;
     float speakerSize = 12.0f;
-    //master level
     float masterLvl = 6.3f;
-    //number of amp modeling presets
     int ampModelingPresets = 99;
-    //fx level
     float fxvl = 4.3f;
 
-
-    //digital signal processing
     void signalProcessing (bool processSignal, int sampleStart, int sampleEnd);
-    //imitate tube amp 
     void imitateTubeAmp (int ampSelectIndex, std::string speakerIr); 
-    //amplifie sound
     void amplifieSound (int inputSig, float gain);
-
 };
 
-/*
-Thing 3) Sampling Pad
-5 properties:
-    1) number of pads
-    2) number of intergrated instruments 
-    3) stereo or mono output 
-    4) amount of sampling  memory space in sec
-    5) number of integrated effects
-3 things it can do:
-    1) trigger  .wav sounds
-    2) play a sequence
-    3) send midi
- */
+void SolideStateAmp::signalProcessing(bool processSignal, int sampleStart, int sampleEnd)
+{
+    if (processSignal != 1)
+    {
+        sampleStart = 0;
+        sampleEnd = 0;
+    }
+    else
+    {
+        sampleStart = 0;
+        sampleEnd = 1;
+    }
+}
+
+void SolideStateAmp::imitateTubeAmp(int ampSelectIndex, std::string speakerIr)
+{
+    if (ampSelectIndex <= 50 && speakerIr == "Celestion")
+    {
+        numSpeakers = 4;
+        speakerSize = 12.0f;
+    }
+    else
+    {
+        numSpeakers = 2;
+        speakerSize = 10.0f; 
+    }
+}
+
+void SolideStateAmp::amplifieSound(int inputSig, float gain)
+{
+    masterLvl = (inputSig * gain)/100;
+}
 
 struct SamplingPad
 {
-    //number of pads
     int numPads = 8;
-    //number of intergrated instruments
     int numIntergratedInst = 25; 
-    //stereo or mono output
     bool stereoOut = true; 
-    //amount of sampling  memory space in sec
     int samplingSpaceSec = 120;
-    //number of integrated effects
     int numIntegratedFx = 12;
+    int padSteps = 16;
 
-    //trigger  .wav sounds
-    void trigSnd (int padIndex, bool trig);
-    //play a sequence
+    void trigSnd ();
     void playSequence (int numSteps, int bpm, bool startSequence);
-    //send midi
-    void sendMidi (int midiNoteNum, int midiVel, bool noteEvent, bool isCc);
+    int sendMidi (int midiNoteNum, int midiVel, bool noteEvent, bool isCc);
 };
 
-/*
-Thing 4) analog console 
-5 properties:
-    1) number of channel strips
-    2) number of audio send
-    3) number of monitor out
-    4) number of steroe chanels
-    5) pre or post EQ selector
-3 things it can do:
-    1) Mix different sounds together 
-    2) contrôl master output level
-    3) add tone
- */
+void SamplingPad::trigSnd(){}
+
+void SamplingPad::playSequence(int numSteps, int bpm, bool startSequence)
+{
+    if (startSequence == 1)
+    {
+        padSteps = numSteps * bpm;
+    }
+    else
+    {
+        padSteps = 0;
+    }
+}
```

Comment:
See if you can rewrite this as a ternary expression.
hint:  Ternary Expressions look like this: `(comparison) ? true expression : false expression ;`
if the whole ternary expression is on the right side of a `=`, after evaluating the `comparison`, your `true/false` expression will be assigned to the left side of the `=`.
- [ ]